src/crons/websocket/blocks.gateway.ts

```diff
@@ -0,0 +1,70 @@
+import { WebSocketGateway, WebSocketServer, SubscribeMessage, MessageBody, ConnectedSocket } from '@nestjs/websockets';
+import { Server, Socket } from 'socket.io';
+import { BlockService } from '../../endpoints/blocks/block.service';
+import { BlockFilter } from '../../endpoints/blocks/entities/block.filter';
+import { QueryPagination } from 'src/common/entities/query.pagination';
+import { BlockSubscribePayload } from '../../endpoints/blocks/entities/block.subscribe';
+import { UseFilters } from '@nestjs/common';
+import { WebsocketExceptionsFilter } from 'src/utils/ws-exceptions.filter';
+import { WsValidationPipe } from 'src/utils/ws-validation.pipe';
+import { OriginLogger } from '@multiversx/sdk-nestjs-common';
+
+@UseFilters(WebsocketExceptionsFilter)
+@WebSocketGateway({ cors: { origin: '*' }, path: '/ws/subscription' })
+export class BlocksGateway {
+  private readonly logger = new OriginLogger(BlocksGateway.name);
+
+  @WebSocketServer()
+  server!: Server;
+
+  constructor(private readonly blockService: BlockService) { }
+
+  @SubscribeMessage('subscribeBlocks')
+  async handleSubscription(
+    @ConnectedSocket() client: Socket,
+    @MessageBody(new WsValidationPipe()) payload: BlockSubscribePayload
+  ) {
+    const filterHash = JSON.stringify(payload);
```

Comment:
this is not really a hash, so maybe we should update the naming to something like `filterIdentifier`. Please apply this suggestion in all the gateways